CONTRIBUTING.md

```diff
@@ -155,9 +145,30 @@ Most of the style guidelines can be checked, fixed, and enforced via Ruff. Older
   - "Fix function foobar"
 - Pull Requests should have an adequate title and description which clearly outline your intentions and changes/additions. Feel free to provide screenshots, GIFs, or videos, especially for UI changes.
 
+#### Adding dependencies
+
+If you want to add a dependency to the project use `poetry add <dependency-name>`.  
+For example: `poetry add httpx`.
+If you decide you did not want to add the dependency after all you can use `poetry remove <dependency-name>`.
+
+If you decide to commit this change be sure to also update the requirements.txt file by generating it from the pyproject.toml with:
+
+```shell
+poetry export --without-hashes --format=requirements.txt > requirements.txt
+```
+
+If you want to add a dev dependency instead use: `poetry add <name> --group dev`. 
+To regenerate the requirements-dev.txt use: 
+
+```shell
+poetry export --without-hashes --without main --with dev --format=requirements.txt > requirements-dev.txt
+```
+
+Do not forget to actually commit the updated requirement.txt and pyproject.toml and poetry.lock files when you change/add dependencies.
+
```

Comment:
There is an important detail missing from this MD file. 

The explanation of what the lock file is. The troublesome command `poetry update` and en explicit warning that this command updates packages and creates a new .lock file. If the PR is updating dependencies, sure, you would use this and update the .lock file.  

When you add a new dependencies, poetry will use the caret symbol, I personally dislike this behaviour as it is implicit in nature. Are future dependencies going to follow the pinned nature of the repo? I don't know, but poetry will update the pyproject.toml file as it sees best. This might lead to trouble  and worth a note in the readme.

Reply:
The section you comment on explains how to add dependencies. The reader does not need to know what the `poetry.lock` file is because it is updated correctly when they run `poetry add <dependency>` as described. No need to run `poetry update` either. I don't understand why you bring up `poetry update` because I did not tell the reader to run this command. 

I could add a sentence saying that it is better to pin a specific version of a dependency `poetry add package==0.1.0`?

Reply:
It is a thing I have seen many times in projects that start using poetry. I don't mind the sections that are already added. Im saying I think this detail is missing.  It has burned many people in projects I have worked on the past years.

Reply:
Can you suggest a text? Like there is not more to it then the poetry.lock pins the versions of each dependency so you get reproducible builds.